src/Protocol/Diaspora.php

```diff
@@ -321,14 +321,14 @@ public static function decodeRaw(string $raw, string $privKey = '', bool $no_exi
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 * @throws \ImagickException
 	 */
-	public static function decode(string $xml, string $privKey = '')
+	public static function decode(string $xml, string $privKey = ''): array
 	{
 		$public = false;
 		$basedom = XML::parseString($xml);
 
 		if (!is_object($basedom)) {
 			Logger::notice('XML is not parseable.');
-			return false;
+			return [];
```

Comment:
Please update the documentation to include the possibility of empty array in return values.

Reply:
Some outside methods rely on returned `false` in case of error. I guess this may need some rewrite later to have returned type-hint `array` back.